docs/usage.md

```diff
@@ -0,0 +1,107 @@
+# Usage
+
+There are two main ways to use nion from React: as a _hook_, or as a _HOC_ (higher-order component). The HOC usage itself comes in two forms: modern HOC and legacy decorators.
+
+## As a hook
+
+When writing new code, the recommended way to use nion is the `useNion` [hook](https://reactjs.org/docs/hooks-intro.html). Note that the `dataKey` in the declaration is required.
+
+```js
+import { useNion } from '@nion/nion'
+
+export const UserContainer = () => {
+    const [currentUser, actions, request] = useNion({
+        dataKey: 'currentUser',
+        endpoint: 'https://patreon.com/api/current_user',
+    })
+
+    const loadButton = <Button onClick={() => actions.get()}>Load</Button>
+
+    return (
+        <Card>
+            {request.isLoading ? <LoadingSpinner /> : loadButton}
+            {currentUser && <UserCard user={currentUser} />}
```

Comment:
I'm guessing `currentUser` here is the same as `this.props.nion.currentUser.data` in HOC+decorator-land? I think it's confusing if the docs sometimes uses `currentUser` interchangeably for the actual data and for the `nion` object at the `dataKey` "`currentUser`"

(That `.data` bit confused a lot of people when we introduced it)

Reply:
Do you think naming something like `const currentUserNion = this.props.nion.currentUser` could help alleviate this confusion?

Reply:
The hook does not have a `.data`, the return from the hook is a thruple: `[data, actions, request]` and those can be named whatever!

Reply:
Right, so:

- When using the hook, as long as you know the order of the triple off the top of your head, it's trivial to rename, so calling the zeroth element `currentUser` is totally fine.

```js
const [currentUser, actions, request] = useNion({ ... })
```

- When using the HOC, `this.props.nion.currentUser` isn't the data, an additional `.data` is needed. Knowing this, how do we feel about the following?

```js
// Alpha: avoid?
const { currentUser } = this.props.nion

// Beta: avoid?
const currentUser = this.props.nion.currentUser

// Charlie: okay?
const currentUserNion = this.props.nion.currentUser
const { data, request, actions } = this.props.nion.currentUser

// Delta: okay?
const { data: currentUser, request, actions } = this.props.nion.currentUser

// Echo: okay?
const { data: currentUser, request: currentUserRequest, actions: currentUserActions } = this.props.nion.currentUser
```

Reply:
We usually do `delta`, but also `echo`. If you have multiple entries you can do:

```
const {
  currentUser: {data: currentUser},
  campaign: {data: campaign},
} = this.props.nion;
```